src/kernels/execution/cellExecution.ts

```diff
@@ -171,7 +171,6 @@ export class CellExecution implements IDisposable {
         // Begin the request that will modify our cell.
         this.execute(this.codeOverride || this.cell.document.getText().replace(/\r\n/g, '\n'), session)
             .catch((e) => this.completedWithErrors(e))
-            .finally(() => this.dispose())
             .catch(noop);
```

Comment:
already handled exceptions, so this catch can be removed?

Reply:
nope, this is to ensure we don't have any floating promises.
else nodejs complains about these promises.
eg things could fail in finally, hence we need a catch after that...